rbac/management/role/v2_model.py

```diff
@@ -0,0 +1,182 @@
+#
+# Copyright 2025 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Model for role V2 management."""
+
+from uuid import uuid4
+
+from django.db import models
+from django.utils import timezone
+from management.models import Permission, Role
+from management.rbac_fields import AutoDateTimeField
+from rest_framework import serializers
+
+from api.models import TenantAwareModel
+
+
+class RoleV2(TenantAwareModel):
+    """V2 Role model."""
+
+    class Types(models.TextChoices):
+        CUSTOM = "custom"
+        SEEDED = "seeded"
+        PLATFORM = "platform"
+
+    uuid = models.UUIDField(default=uuid4, editable=False, unique=True, null=False)
```

Comment:
We started using `uuid7` in the Workspace model: https://github.com/RedHatInsights/insights-rbac/blob/86e67c8d53cb044abfff49e26c23157b1682d350/rbac/management/workspace/model.py#L46 since it's time-based and thus creates better index/db performance. We should consider using it for all new models.

Reply:
updated